src/main/java/seedu/taskman/model/event/Activity.java

```diff
@@ -0,0 +1,132 @@
+package seedu.taskman.model.event;
+
+import seedu.taskman.model.tag.UniqueTagList;
+
+import java.util.Objects;
+import java.util.Optional;
+
+/**
+ * Wrapper for both ReadOnlyEvent and ReadOnlyTask
+ */
+public class Activity implements ReadOnlyEvent{
```

Comment:
Tried implemented ReadOnlyTask at first but was not sure how to deal with getStatus since it isnt a field present in Event. Returning Optional<Status> for getStatus makes more sense for Activity and changing ReadOnlyTask to adapt to Activity seems a bit odd too.